src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs

```diff
@@ -47,15 +48,80 @@ internal class SdkResolverLoader
         /// </summary>
         /// <param name="rootFolder"></param>
         /// <returns></returns>
-        internal virtual IList<string> FindPotentialSdkResolvers(string rootFolder)
+        internal virtual IList<string> FindPotentialSdkResolvers(string rootFolder, ElementLocation location)
         {
+            var assembliesList = new List<string>();
+
             if (string.IsNullOrEmpty(rootFolder) || !FileUtilities.DirectoryExistsNoThrow(rootFolder))
-                return new List<string>();
+            {
+                return assembliesList;
+            }
+
+            foreach (var subfolder in new DirectoryInfo(rootFolder).GetDirectories())
+            {
+                var assembly = Path.Combine(subfolder.FullName, $"{subfolder.Name}.dll");
+                var manifest = Path.Combine(subfolder.FullName, $"{subfolder.Name}.xml");
+
+                var assemblyAdded = TryAddAssembly(assembly, assembliesList);
+                if (!assemblyAdded)
+                {
+                    assemblyAdded = TryAddAssemblyFromManifest(manifest, subfolder.FullName, assembliesList, location);
+                }
+
+                if (!assemblyAdded)
+                {
+                    ProjectFileErrorUtilities.ThrowInvalidProjectFile(new BuildEventFileInfo(location), "SdkResolverNoDllOrManifest", subfolder.FullName);
+                }
+            }
+
+            return assembliesList;
+        }
+
+        private bool TryAddAssemblyFromManifest(string pathToManifest, string manifestFolder, List<string> assembliesList, ElementLocation location)
+        {
+            if (!string.IsNullOrEmpty(pathToManifest) && !FileUtilities.FileExistsNoThrow(pathToManifest)) return false;
+
+            string path = null;
+
+            try
+            {
+                // <SdkResolver>
+                //   <Path>...</Path>
+                // </SdkResolver>
+                var manifest = SdkResolverManifest.Load(pathToManifest);
+
+                if (manifest == null || string.IsNullOrEmpty(manifest.Path))
+                {
+                    ProjectFileErrorUtilities.ThrowInvalidProjectFile(new BuildEventFileInfo(location), "SdkResolverDllInManifestMissing", pathToManifest, string.Empty);
+                }
+
+                path = manifest.Path;
+            }
+            catch (SerializationException e)
+            {
+                ProjectFileErrorUtilities.ThrowInvalidProjectFile(new BuildEventFileInfo(location), e, "SdkResolverManifestInvalid", pathToManifest, e.Message);
```

Comment:
Isn't it usually better to pass `e.ToString()` rather than `Message`? IIRC that tends to have more information.

Reply:
This will lose stacktraces, but they're not very useful. The message in this case is really specific (invalid character in the xml on line X column Y, etc.), The stack trace would just be in the deserialize method.

Reply:
Good enough for me. Maybe comment that here?